### ISSUE TITLE
Unable to compile the source

### ISSUE THREAD

**Issue by ziotom78** [amoradinejad/limHaloPT#5]
I am currently struggling to compile the package, because of the global variable `gb`:

```
$ make
cc -g -c -fopenmp main.c -o main.o -I../Include/ -I/opt/local/include  -I../Class/include  -I/home/tomasi/tmp/include 
cc -g -c -fopenmp hcubature.c -o cubature.o -I../Include/ -I/opt/local/include  -I../Class/include  -I/home/tomasi/tmp/include
cc -g -c -fopenmp utilities.c -o utilities.o -I../Include/ -I/opt/local/include  -I../Class/include  -I/home/tomasi/tmp/include
cc -g -c -fopenmp cosmology.c -o cosmology.o -I../Include/ -I/opt/local/include  -I../Class/include  -I/home/tomasi/tmp/include
cc -g -c -fopenmp survey_specs.c -o survey_specs.o -I../Include/ -I/opt/local/include  -I../Class/include  -I/home/tomasi/tmp/include	
cc -g -c  -fopenmp line_ingredients.c -o line_ingredients.o -I../Include/ -I/opt/local/include  -I../Class/include  -I/home/tomasi/tmp/include 
cc -g -c  -fopenmp wnw_split.c -o wnw_split.o -I../Include/ -I/opt/local/include  -I../Class/include  -I/home/tomasi/tmp/include
cc -g -c  -fopenmp ir_res.c -o ir_res.o -I../Include/ -I/opt/local/include  -I../Class/include  -I/home/tomasi/tmp/include
cc -g -c  -fopenmp ps_halo_1loop.c -o ps_halo_1loop.o -I../Include/ -I/opt/local/include  -I../Class/include  -I/home/tomasi/tmp/include
cc -g -c  -fopenmp ps_line_pt.c -o ps_line_pt.o -I../Include/ -I/opt/local/include  -I../Class/include  -I/home/tomasi/tmp/include
cc -g -c  -fopenmp ps_line_hm.c -o ps_line_hm.o -I../Include/ -I/opt/local/include  -I../Class/include  -I/home/tomasi/tmp/include
cc main.o cubature.o utilities.o cosmology.o survey_specs.o line_ingredients.o wnw_split.o ir_res.o ps_halo_1loop.o ps_line_pt.o ps_line_hm.o  -fopenmp -Wall -I../Include/ -I/opt/local/include  -I../Class/include  -I/home/tomasi/tmp/include -L/opt/local/lib  -L../Class/lib -L/home/tomasi/tmp/lib/  -lgsl -lgslcblas -lcuba  -lclass -lm -o limHaloPT       
/usr/bin/ld: cosmology.o:/home/tomasi/Documents/work/articles/referee/JOSS/limHaloPT/Source/cosmology.c:51: multiple definition of `gb'; main.o:/home/tomasi/Documents/work/articles/referee/JOSS/limHaloPT/Source/main.c:13: first defined here
/usr/bin/ld: survey_specs.o:/home/tomasi/Documents/work/articles/referee/JOSS/limHaloPT/Source/survey_specs.c:14: multiple definition of `gb'; main.o:/home/tomasi/Documents/work/articles/referee/JOSS/limHaloPT/Source/main.c:13: first defined here
/usr/bin/ld: line_ingredients.o:/home/tomasi/Documents/work/articles/referee/JOSS/limHaloPT/Source/line_ingredients.c:33: multiple definition of `gb'; main.o:/home/tomasi/Documents/work/articles/referee/JOSS/limHaloPT/Source/main.c:13: first defined here
/usr/bin/ld: ir_res.o:/home/tomasi/Documents/work/articles/referee/JOSS/limHaloPT/Source/ir_res.c:20: multiple definition of `gb'; main.o:/home/tomasi/Documents/work/articles/referee/JOSS/limHaloPT/Source/main.c:13: first defined here
/usr/bin/ld: ps_halo_1loop.o:/home/tomasi/Documents/work/articles/referee/JOSS/limHaloPT/Source/ps_halo_1loop.c:25: multiple definition of `gb'; main.o:/home/tomasi/Documents/work/articles/referee/JOSS/limHaloPT/Source/main.c:13: first defined here
/usr/bin/ld: ps_line_pt.o:/home/tomasi/Documents/work/articles/referee/JOSS/limHaloPT/Source/ps_line_pt.c:20: multiple definition of `gb'; main.o:/home/tomasi/Documents/work/articles/referee/JOSS/limHaloPT/Source/main.c:13: first defined here
/usr/bin/ld: ps_line_hm.o:/home/tomasi/Documents/work/articles/referee/JOSS/limHaloPT/Source/ps_line_hm.c:21: multiple definition of `gb'; main.o:/home/tomasi/Documents/work/articles/referee/JOSS/limHaloPT/Source/main.c:13: first defined here
collect2: error: ld returned 1 exit status
make: *** [makefile:8: limHaloPT] Error 1
```

It seems that the variable `globals gb` is being declared either as

```c
extern struct globals gb;
```

or

```c
struct globals gb;
```

I believe that there should just be a declaration

```c
extern struct globals gb;
```

in **one** of the header files (e.g., at the bottom of `header.h`) and a definition

```c
struct globals gb;
```

in **one** of the source files (e.g., `main.c`).

**Comment by ziotom78** [comment]
> As suggested, all instances of declaration of gb are removed, and now it is only declared in header.h to fix #5.

Mmm, I still have the same problems compiling the code. I see that you removed some instances, but there are still others that prevent the code from compiling with GCC 11:

- [`Source/ir_res.c`](https://github.com/amoradinejad/limHaloPT/blob/master/Source/ir_res.c#L20), line 20
- [`Source/line_ingredients.c`](https://github.com/amoradinejad/limHaloPT/blob/master/Source/line_ingredients.c#L33), line 33
- [`Source/ps_halo_1loop.c`](https://github.com/amoradinejad/limHaloPT/blob/master/Source/ps_halo_1loop.c#L25), line 25
- [`Source/ps_line_hm.c`](https://github.com/amoradinejad/limHaloPT/blob/master/Source/ps_line_hm.c#L21), line 21
- [`Source/ps_line_pt.c`](https://github.com/amoradinejad/limHaloPT/blob/master/Source/ps_line_pt.c#L20), line 20
- [`Source/survey_specs.c`](https://github.com/amoradinejad/limHaloPT/blob/master/Source/survey_specs.c#L14), line 14

plus of course the definition in `Source/main.c` which is the only one that should survive.

**Comment by amoradinejad** [comment]
All instances of declaration of gb, except in main.c are removed.
